### ISSUE TITLE
Choosing one of multiple Vault instances to fetch secrets from

### ISSUE THREAD

**Issue by aiman-alsari** [hashicorp/vault-k8s#299]
**Describe the bug**

Under the [documentation](https://www.vaultproject.io/docs/platform/k8s/injector/annotations#vault-annotations) there is a comment that states

> Vault annotations change how the Vault Agent containers communicate with Vault. For example, **Vault's address**, TLS certificates to use, client parameters such as timeouts, etc.

As far as I can see by looking at both the annotations defined lower down and in the code, that statement is incorrect. There is no way to override the VAULT_ADDR that this injector deals with.

I wasn't sure whether to make this a feature request or a documentation bug, but let's start with the docs issue first.

**Comment by tvoran** [comment]
Hi @aiman-alsari, have you tried the [`vault.hashicorp.com/service` annotation](https://www.vaultproject.io/docs/platform/k8s/injector/annotations#vault-hashicorp-com-service)? I think it should do what you're looking for.

**Comment by aiman-alsari** [comment]
@tvoran yes you are indeed correct, that setting is what I am looking for.
My initial thought was that it was for choosing the k8s service that Vault was running on inside the cluster, and my Vault was running outside of the cluster so I assumed it didn't apply.

**Comment by aiman-alsari** [comment]
There's a slightly confusing discrepancy in terminology used here, `service` implies kubernetes service, `addr` is what is used everywhere else.

**Comment by tvoran** [comment]
Yes, that terminology is a bit confusing. My guess is it's because the external vault case wasn't considered when that annotation was added. I'll update the docs for it at least, and we should consider renaming that annotation in the future too.
